restcomm/restcomm.ussd/src/main/java/org/restcomm/connect/ussd/telephony/UssdCallManager.java

```diff
@@ -127,6 +130,8 @@ public UssdCallManager(Configuration configuration, ServletContext context, SipF
         numberSelector = (NumberSelectorService)context.getAttribute(NumberSelectorService.class.getName());
 
         extensions = ExtensionController.getInstance().getExtensions(ExtensionType.FeatureAccessControl);
+
+        uriUtils = RestcommConnectServiceProvider.getInstance().uriUtils();
```

Comment:
see my previous comments about injecting dependencies in a static manner like this. Hurts testability.

Reply:
@hrosa Earlier you mentioned ServletContext but ServletContext is not available in the UssdCallManager and many other places in the project.

So can you elaborate on what do you suggest? Can you provide a concrete plan for how you think we should deal with this?

My point of view is that `RestcommConnectServiceProvider` should not be singleton as it was first created, but this a major refactoring that cannot be part of the story here.